Exercise_4.py

```diff
@@ -1,57 +1,57 @@
 # Python program to insert element in binary tree  
 class newNode():  
   
-    def __init__(self, data):  
-        self.key = data 
-        self.left = None
-        self.right = None
-          
+	def __init__(self, data):  
+		self.key = data 
+		self.left = None
+		self.right = None
+		  
 """ Inorder traversal of a binary tree"""
 def inorder(temp): 
-    if temp:
-        inorder(temp.left)
-        print(temp.key)
-        inorder(temp.right)
+	if temp:
+		inorder(temp.left)
+		print(temp.key)
+		inorder(temp.right)
   
   
 """function to insert element in binary tree """
 def insert(temp,key):
-    if not temp:
-        temp = newNode(key)
-    else:
-        if not temp.left:
-            temp.left = newNode(key)
-            return
-        elif temp.left:
-            insert(temp.left, key)
-            return
-        if not temp.right:
-            temp.right = newNode(key)
-            return
-        elif temp.right:
-            insert(temp.right, key)
-            return
+	if not temp:
+		temp = newNode(key)
+	else:
+		if not temp.left:
+			temp.left = newNode(key)
+			return
+		elif temp.left:
+			insert(temp.left, key)
+			return
+		if not temp.right:
+			temp.right = newNode(key)
+			return
+		elif temp.right:
+			insert(temp.right, key)
+			return
 
 #I failed to understand the condition to insert a node in the binary tree since there is no ordering as such. Therefore it is inserted as the left of 7
   
-    
+	
   
-       
+	   
```

Comment:
refer this code for a level order traversal of a tree    
static void insert(Node temp, int key) 
    { 
        // Do level order traversal until we find 
        // an empty place and add the node.  
        Queue<Node> q = new LinkedList<>();
        q.add(temp);
        while (!q.isEmpty()) {
            int size = q.size();
            for (int i = 0; i < size; i++) {
                Node curr = q.poll();
                if (curr.left == null) {
                    curr.left = new Node(key);
                    break;
                }
                if (curr.right == null) {
                    curr.right = new Node(key);
                    break;
                }
                if (curr.left != null) q.add(curr.left);
                if (curr.right != null) q.add(curr.right);
            }
        }
    }